src/system_program.rs

```diff
@@ -85,7 +89,7 @@ impl SystemProgram {
                 }
                 SystemProgram::Assign { program_id } => {
                     if !Self::check_id(&accounts[0].program_id) {
-                        Err(Error::InvalidArgument)?;
+                        Err(Error::AssignOfUnownedAccount)?;
```

Comment:
AssignOfOwnedAccount no?

Reply:
I read this as "Not owned by SystemProgram so SystemProgram cannot assign"